cli/src/utils.rs

```diff
@@ -19,3 +24,84 @@ pub fn format_devices(devices: &[AvailableDevice]) {
         println!("{YELLOW}{slug}{RESET} - {organization_id}: {human_handle} @ {device_label}");
     }
 }
+
+pub async fn load_device_and_run<F, Fut>(
+    config_dir: Option<PathBuf>,
+    device_slughash: Option<String>,
+    function: F,
+) -> anyhow::Result<()>
+where
+    F: FnOnce(AuthenticatedCmds, Arc<LocalDevice>) -> Fut,
+    Fut: Future<Output = anyhow::Result<()>>,
+{
+    let config_dir = config_dir.unwrap_or(get_default_config_dir());
+    let devices = list_available_devices(&config_dir).await;
+
+    if let Some(device_slughash) = device_slughash {
+        let mut possible_devices = vec![];
+
+        for device in &devices {
+            if device.slughash().starts_with(&device_slughash) {
+                possible_devices.push(device);
+            }
+        }
+
+        match possible_devices.len() {
+            0 => {
+                println!("Device `{device_slughash}` not found, available devices:");
+                format_devices(&devices);
+            }
+            1 => {
+                let device = &possible_devices[0];
+
+                let device = match device.ty {
+                    DeviceFileType::Password => {
+                        #[cfg(feature = "testenv")]
+                        let password = "test".to_string().into();
+                        #[cfg(not(feature = "testenv"))]
+                        let password = rpassword::prompt_password("password:")?.into();
+
+                        let access = DeviceAccessStrategy::Password {
+                            key_file: device.key_file_path.clone(),
+                            password,
+                        };
+
+                        // The password is invalid or the binary is compiled with fast crypto
+                        load_device(&config_dir, &access).await?
```

Comment:
I still do not understand what this comment is about. You answered [here](https://github.com/Scille/parsec-cloud/pull/5818#discussion_r1429894534):
> if you didn't work on libparsec_crypto, you can't know that your password can be erroned due to this crate

What does it means that the password can be erroneous due to `libparsec_crypto`?
What does it have to do with the `load_device` call? the function can return an error if password _is_ invalid?
In any case, that's not what the comment says.

We should re-write the comment so it _helps understand_ the code (specially if there are not-so-obvious implications).

Reply:
What I mean is: the `password` can be `good`, but the error will say that the `password` is `wrong`, that's why I added this comment

Reply:
OK, so that should probably be documented in the `load_device` function or the `libparsec_crypto` (I couldn't find any reference) 🫤

Anyway, what do you think about changing the comment to:
```suggestion
                        // This will fail if the password is invalid, but also if the binary is compiled with fast crypto (see  libparsec_crypto)
                        load_device(&config_dir, &access).await?
```

Reply:
I agree with you